src/components/dialogHelper/dialogHelper.js

```diff
@@ -86,10 +88,14 @@ import '../../assets/css/scrollstyles.scss';
                 document.body.classList.remove('noScroll');
             }
 
-            if (!self.closedByBack && isHistoryEnabled(dlg)) {
-                const state = window.history.state || {};
-                if (state.dialogId === hash) {
-                    appRouter.back();
+            if (isHistoryEnabled(dlg)) {
+                const state = history.location.state || {};
+                if (state.dialogs?.length > 0) {
+                    if (state.dialogs[state.dialogs.length - 1] === hash) {
+                        history.back();
+                    } else if (state.dialogs.includes(hash)) {
+                        console.info('[dialogHelper] dialog "%s" was closed, but is not the last dialog opened', hash);
```

Comment:
Maybe `warn`?

This shouldn't happen, but shouldn't the dangling hash be removed? - If `removeOnClose: true` (most of the time it is), the dialog is automatically removed from DOM.

Reply:
Yeah good call. I have updated it to warn and replace the current history state to exclude the closed dialog hash.